Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/IdentitiesRepository.cs

```diff
@@ -81,10 +81,10 @@ public async Task<DbPaginationResult<Device>> FindAllDevicesOfIdentity(IdentityA
 
     public async Task<Device> GetDeviceById(DeviceId deviceId, CancellationToken cancellationToken, bool track = false)
     {
-        return await (track ? _devices : _readonlyDevices)
+        return (await (track ? _devices : _readonlyDevices)
             .NotDeleted()
             .IncludeAll(_dbContext)
-            .FirstOrDefaultAsync(d => d.Id == deviceId, cancellationToken) ?? throw new Exception($"Device with ID '{deviceId}' not found.");
+            .FirstOrDefaultAsync(d => d.Id == deviceId, cancellationToken))!;
```

Comment:
Throwing exception here makes sense to me but it doesn't work and I couldn't figure out why.